layouts/_default/list.html

```diff
@@ -32,7 +32,11 @@
   <div class="content container-fluid" id="content">
     <div class="container-fluid post-card-holder" id="post-card-holder">
       {{ $posts := where .RegularPagesRecursive "Layout" "!=" "search" }}
-      {{ $paginator := .Paginate $posts 12 }}
+      {{ $numShow := 12}}
+      {{ if site.Params.postsPage.maxPostsPerPage}}
```

Comment:
Let's call the settings `pagination` and put it under `features`. So, the configuration will look like:
```yaml
params:
  features:
    pagination:
       maxPostsPerPage: 10
```

Reply:
You can use [default](https://gohugo.io/functions/compare/default/) function to simplify the code little bit. For example:

```
{{ $numShow = site.Params.features.pagination.maxPostsPerPage | default 12}}
```